### PR TITLE
[DOCS] Add 8.0.0-rc1 release notes

### PR DESCRIPTION
**Pull request by arteam**
## Preview

https://elasticsearch_82469.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/release-notes-8.0.0-rc1.html

### CODE REVIEW

File: docs/reference/release-notes/8.0.0-rc1.asciidoc [hunk 1]
```diff
@@ -8,6 +8,73 @@ coming::[8.0.0-rc1]
...
+=== Deprecations
+
+Engine::
+* Deprecate setting `max_merge_at_once_explicit` {es-pull}80574[#80574]
+
+Machine Learning::
+* Deprecate `estimated_heap_memory_usage_bytes` and replace with `model_size_bytes` {es-pull}80554[#80554]
+
+Search::
+* Configure `IndexSearcher.maxClauseCount()` based on Node characteristics {es-pull}81525[#81525] (issue: {es-issue}46433[#46433])
```
Comment (jrodewig) [inline docs/reference/release-notes/8.0.0-rc1.asciidoc:32]:
```suggestion
* Configure `IndexSearcher.maxClauseCount()` based on node characteristics {es-pull}81525[#81525] (issue: {es-issue}46433[#46433])
```

File: docs/reference/release-notes/8.0.0-rc1.asciidoc [hunk 2]
```diff
@@ -8,6 +8,73 @@ coming::[8.0.0-rc1]
...
+
+Transform::
+* Improve transform deprecation messages {es-pull}81847[#81847] (issues: {es-issue}81521[#81521], {es-issue}81523[#81523])
+
+[[enhancement-8.0.0-rc1]]
+[float]
+=== Enhancements
+
+Authorization::
+* Granting `kibana_system` reserved role access to "all" privileges to .internal.preview.alerts* index {es-pull}80889[#80889] (issues: {es-issue}76624[#76624], {es-issue}80746[#80746], {es-issue}116374[#116374])
```
Comment (jrodewig) [inline docs/reference/release-notes/8.0.0-rc1.asciidoc:42]:
```suggestion
* Granting `kibana_system` reserved role access to "all" privileges to `.internal.preview.alerts*` index {es-pull}80889[#80889] (issues: {es-issue}76624[#76624], {es-issue}80746[#80746], {es-issue}116374[#116374])
```

File: docs/reference/release-notes/8.0.0-rc1.asciidoc [hunk 3]
```diff
@@ -8,6 +8,73 @@ coming::[8.0.0-rc1]
...
+
+ILM+SLM::
+* Make unchanged ILM policy updates into noop {es-pull}82240[#82240] (issue: {es-issue}82065[#82065])
+
+Indices APIs::
+* Batch rollover cluster state updates {es-pull}79945[#79945] (issues: {es-issue}77466[#77466], {es-issue}79782[#79782])
+* Reuse `MappingMetadata` instances in Metadata class {es-pull}80348[#80348] (issues: {es-issue}69772[#69772], {es-issue}77466[#77466])
+
+Infra/Settings::
+* Implement Setting Deduplication via String Interning {es-pull}80493[#80493] (issues: {es-issue}77466[#77466], {es-issue}78892[#78892])
```
Comment (jrodewig) [inline docs/reference/release-notes/8.0.0-rc1.asciidoc:54]:
```suggestion
* Implement setting deduplication via string interning {es-pull}80493[#80493] (issues: {es-issue}77466[#77466], {es-issue}78892[#78892])
```

File: docs/reference/release-notes/8.0.0-rc1.asciidoc [hunk 4]
```diff
@@ -8,6 +8,73 @@ coming::[8.0.0-rc1]
...
+
+Machine Learning::
+* Add `deployment_stats` to trained model stats {es-pull}80531[#80531]
+* Use_auto_machine_memory_percent now defaults `max_model_memory_limit` {es-pull}80532[#80532] (issue: {es-issue}80415[#80415])
+
+Monitoring::
+* Adding default templates for Metricbeat ECS data {es-pull}81744[#81744]
+
+Network::
+* Improve Slow Inbound Handling to Include Response Type {es-pull}80425[#80425]
```
Comment (jrodewig) [inline docs/reference/release-notes/8.0.0-rc1.asciidoc:67]:
```suggestion
* Improve slow inbound handling to include response type {es-pull}80425[#80425]
```

File: docs/reference/release-notes/8.0.0-rc1.asciidoc [hunk 5]
```diff
@@ -8,6 +8,73 @@ coming::[8.0.0-rc1]
...
+
+Recovery::
+* Fix `PendingReplicationActions` Submitting lots of NOOP Tasks to GENERIC {es-pull}82092[#82092] (issues: {es-issue}77466[#77466], {es-issue}79837[#79837])
```
Comment (jrodewig) [inline docs/reference/release-notes/8.0.0-rc1.asciidoc:70]:
```suggestion
* Fix `PendingReplicationActions` submitting lots of `NOOP` tasks to `GENERIC` {es-pull}82092[#82092] (issues: {es-issue}77466[#77466], {es-issue}79837[#79837])
```

**Review by jrodewig — APPROVED** [review]
LGTM with a few fixes for grammar nits.

@lcawl @szabosteve Would one of you take a look at the ML-related items? It looks like there were several added with this PR that weren't included in https://github.com/elastic/elasticsearch/pull/82390.

**Review by szabosteve — APPROVED** [review]
ML-related items LGTM!

**Comment by jrodewig** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------:|
|✅|8.0|[](https://github.com/elastic/elasticsearch/pull/82473)|

**Pull request merged**